ddtrace/tracer/textmap.go

```diff
@@ -335,14 +347,28 @@ func (p *chainedPropagator) Extract(carrier interface{}) (*SpanContext, error) {
 				links = append(links, link)
 			}
 		}
+	}
 
-		if _, ok := v.(*propagatorBaggage); ok && extractedCtx != nil {
-			if len(extractedCtx.baggage) > 0 {
-				ctx.baggage = extractedCtx.baggage
-				atomic.StoreUint32(&ctx.hasBaggage, 1)
-			}
+	if ctx == nil && len(pendingBaggage) > 0 {
+		rootSpan := spanStart("baggage-root")
+		ctx = rootSpan.context
+		if ctx.baggage == nil {
+			ctx.baggage = make(map[string]string, len(pendingBaggage))
+		}
+		for k, v := range pendingBaggage {
+			ctx.baggage[k] = v
+		}
+		atomic.StoreUint32(&ctx.hasBaggage, 1)
+	}
```

Comment:
Starting this span can have unexpected consequences. We are extracting a context, this code shouldn't create spans.

The extracted context contains a new trace with the fake span as part of it. Any span started from this context will show as disconnected from the trace that generated the request to the instrumented service.

Is it possible that we get a trace ID from the carrier when using exclusively `baggage`?